packages/vite/src/node/config.ts

```diff
@@ -477,12 +477,34 @@ export async function resolveConfig(
   const resolvedRoot = normalizePath(
     config.root ? path.resolve(config.root) : process.cwd(),
   )
+
+  // We will warn if the root path includes either the '#' or '?' character
+  const badPathCharacters = []
```

Comment:
Since this logic is a little big now, can we move it as a function?

Reply:
Sounds reasonable 👍